File: javascript/primes/Simple.js
#!/usr/bin/env node

'use strict';

const NUMBER = 100000;

function getLastPrime(number)
{
    var lastPrime = 2;

    // Traverse each number from 1 to N with the help of for loop
    for (var i = 2; i <= number; i++)
    {
        var isPrime = true;

        for (var j = 2; j <= i / 2; ++j) {
            if (i % j == 0) {
                isPrime = false;
                break;
            }
        }

        if (isPrime) {
            lastPrime = i;
        }
    }

    return lastPrime;
}

(async function() {
    const startTimeMs = new Date().getTime();
      
    const lastPrime = getLastPrime(NUMBER);
    console.log(`Last prime: ${lastPrime}`);

    const endTimeMs = new Date().getTime();
    const durationMs = endTimeMs - startTimeMs;
    console.log(`Execution time: ${durationMs}ms`);
})();

File: swift/primes/Simple.swift
#!/usr/bin/swift

import Foundation

let NUMBER = 100000

func getLastPrime(_ count: Int) -> Int {
    var lastPrime = 2

    for i in 2...count {
        var isPrime = true

        for j in 2...(i / 2 + 1) {
            if i % j == 0 {
                isPrime = false
                break
            }
        }

        if isPrime {
            lastPrime = i
        }
    }

    return lastPrime
}

let startTimeMs = Int(Date().timeIntervalSince1970 * 1000)

let lastPrime = getLastPrime(NUMBER)
print("Last prime: \(lastPrime)")

let endTimeMs = Int(Date().timeIntervalSince1970 * 1000)
let durationMs = endTimeMs - startTimeMs

print("Execution time: \(durationMs)ms")
